include/seqan3/utility/type_inspection.hpp

```diff
@@ -0,0 +1,68 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides traits to inspect some information of a type, for example its name.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#if defined(__GNUC__) || defined(__clang__)
+#include <cxxabi.h>
+#endif // defined(__GNUC__) || defined(__clang__)
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <typeinfo>
+
+#include <seqan3/core/platform.hpp>
+
+namespace seqan3::detail
```

Comment:
Now the file is not in the detail folder anymore but the entity is still in the detail namespace?

Reply:
Yes I think for now that's best. We can still decide if we want to make it public later on. (I don't remember if we explicitly decided to make this public?)